src/OpenFeature/Api.cs

```diff
@@ -127,7 +127,26 @@ public FeatureProvider GetProvider(string clientName)
         /// </para>
         /// </summary>
         /// <param name="hooks">A list of <see cref="Hook"/></param>
-        public void AddHooks(IEnumerable<Hook> hooks) => this._hooks.PushRange(hooks.ToArray());
+        public void AddHooks(IEnumerable<Hook> hooks)
+#if NET7_0_OR_GREATER
+            => this._hooks.PushRange(hooks as Hook[] ?? hooks.ToArray());
+#else

```

Comment:
I guess the only reason for the use of this directive is to save use the additional checks? I suppose we could always just run the `#else` block?

I wouldn't say this is a hot code path, so I'm not sure if it's worth it, but I don't feel strongly about it.

Reply:
I agree, but it's not so much the hot path I was thinking about here, but rather flagging that this behavior can get 🪓'd in the future when our TFMs are all `NET7_0_OR_GREATER` (which theoretically will happen...someday).

Without the `#if`/`#else` I'd worry that some future reader might misunderstand what we're doing here (especially since this is super easy to miss if you're only running the default `net8.0` tests locally).

Reply:
Good enough for me.